pulp_container/tests/functional/api/test_push_content.py

```diff
@@ -16,313 +17,345 @@
+def test_push_matching_username(
+    add_to_cleanup,
+    gen_user,
+    registry_client,
+    local_registry,
+    container_distribution_api,
+    container_namespace_api,
+):
+    """
+    Test that you can push to a nonexisting namespace that matches your username.
+    """
+    user_helpless = gen_user()
+    namespace_name = user_helpless.username
+    repo_name = f"{namespace_name}/matching"
+    local_url = f"{repo_name}:2.0"
+    invalid_local_url = f"other/{repo_name}:2.0"
+    image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_a"
+
+    registry_client.pull(image_path)
+    with user_helpless:
+        local_registry.push(image_path, local_url)
+        namespace = container_namespace_api.list(name=namespace_name).results[0]
+        add_to_cleanup(container_namespace_api, namespace.pulp_href)
+
+        with pytest.raises(exceptions.CalledProcessError):
+            local_registry.push(image_path, invalid_local_url)
 
         # test you can create distribution under the namespace that matches login
         repo_name2 = f"{namespace_name}/matching2"
         distribution = {"name": repo_name2, "base_path": repo_name2, "private": True}
-        distribution_response = self.user_helpless["distribution_api"].create(distribution)
+        distribution_response = container_distribution_api.create(distribution)
         created_resources = monitor_task(distribution_response.task).created_resources
-        distribution = self.user_helpless["distribution_api"].read(created_resources[0])
+        distribution = container_distribution_api.read(created_resources[0])
 
         # cleanup, namespace removal also removes related distributions
-        namespace = self.namespace_api.list(name=namespace_name).results[0]
-        namespace_response = self.namespace_api.delete(namespace.pulp_href)
+        namespace_response = container_namespace_api.delete(namespace.pulp_href)
         monitor_task(namespace_response.task)
 
         # test you can create distribution if namespace does not exist but matches login
         distribution = {"name": repo_name, "base_path": repo_name, "private": True}
-        distribution_response = self.user_helpless["distribution_api"].create(distribution)
+        distribution_response = container_distribution_api.create(distribution)
         created_resources = monitor_task(distribution_response.task).created_resources
-        distribution = self.user_helpless["distribution_api"].read(created_resources[0])
+        distribution = container_distribution_api.read(created_resources[0])
 
-        # cleanup, namespace removal also removes related distributions
-        namespace = self.namespace_api.list(name=namespace_name).results[0]
-        self.addCleanup(self.namespace_api.delete, namespace.pulp_href)
+        add_to_cleanup(container_namespace_api, distribution.namespace)
 
 
 class PushManifestListTestCase(PulpTestCase, rbac_base.BaseRegistryTest):
```

Comment:
not sure this test belongs here? we only test with admin perms so maybe it should be moved out?

Reply:
To be fair, the file is not called `test_rbac_push_content`.